x-pack/plugins/fleet/server/services/agents/actions.ts

```diff
@@ -34,7 +36,10 @@ export async function createAgentAction(
   const timestamp = new Date().toISOString();
   const body: FleetServerAgentAction = {
     '@timestamp': timestamp,
-    expiration: newAgentAction.expiration ?? new Date(Date.now() + ONE_MONTH_IN_MS).toISOString(),
+    expiration:
+      newAgentAction.expiration === NO_EXPIRATION
```

Comment:
added back no expiration on upgrade action based on the discussion [here](https://github.com/elastic/kibana/issues/146018#issuecomment-1370730471)

Reply:
I am wondering if we should add back some expiration (e.g. 1 year) so that the Agent logic doesn't fail on missing expiration here: https://github.com/elastic/fleet-server/pull/2243#issuecomment-1372322363

Reply:
added 1y expiration, it seems to work well without having to change the agent code